stream-chat-android-core/src/main/java/io/getstream/chat/android/client/call/Call.kt

```diff
@@ -71,8 +71,11 @@ public interface Call<T : Any> {
  *
  * Does not throw exceptions. Any errors will be wrapped in the [Result] that's returned.
  */
-public suspend fun <T : Any> Call<T>.await(): Result<T> =
-    suspendCancellableCoroutine { continuation ->
+public suspend fun <T : Any> Call<T>.await(): Result<T> {
+    if (this is CoroutineCall<T>) {
+        return this.runnable.invoke()
+    }
```

Comment:
To avoid calling `enqueue` below which then internally would just kick off yet another coroutine in the case of a `CoroutineCall`...